tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -124,6 +125,8 @@ bool IsHLSLNumericUserDefinedType(clang::QualType type) {
 // which can't be annotated. But includes UDTs of trivially copyable data and
 // the builtin trivially copyable raytracing structs.
 bool IsHLSLCopyableAnnotatableRecord(clang::QualType QT) {
+  if (ContainsHitObject(QT))
+    return false;
```

Comment:
I don't know why this special case code is necessary.  `IsHLSLNumericUserDefinedType` should return false for anything containing a HitObject.  If it doesn't, that's a bug that should be fixed there.

Reply:
Removed that special case